xarray/core/groupby.py

```diff
@@ -338,6 +356,11 @@ def lookup_order(dimension):
         new_order = sorted(stacked.dims, key=lookup_order)
         return stacked.transpose(*new_order)
 
+    def _restore_multiindex(self, combined):
+        if self._stacked_dim is not None and self._stacked_dim in combined.dims:
+            combined[self._stacked_dim] = self.group[self._stacked_dim]
```

Comment:
is where we somehow modify the original object being grouped? maybe better to try using `assign` instead of mutation